google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreOptions.java

```diff
@@ -216,8 +221,74 @@ public FirestoreOptions build() {
         }
       }
 
+      // Override credentials and channel provider if we are using the emulator.
+      String emulatorHost = System.getenv(FIRESTORE_EMULATOR_SYSTEM_PROPERTY);
+      if (emulatorHost != null) {
+        String hostUrlString = "http://" + emulatorHost;
+
+        // Try creating a host in order to validate that the host name is valid.
+        try {
+          URI.create(hostUrlString);
+        } catch (Exception e) {
+          throw new IllegalArgumentException(
+              "Value: '"
+                  + emulatorHost
+                  + "' for property "
+                  + FIRESTORE_EMULATOR_SYSTEM_PROPERTY
+                  + " is not a valid host",
+              e);
+        }
+
+        // Setting the channel provider to use plaintext turns off SSL.
+        this.setChannelProvider(
+            InstantiatingGrpcChannelProvider.newBuilder()
+                .setEndpoint(emulatorHost)
+                .setChannelConfigurator(
+                    new ApiFunction<ManagedChannelBuilder, ManagedChannelBuilder>() {
+                      @Override
+                      public ManagedChannelBuilder apply(ManagedChannelBuilder input) {
```

Comment:
any chance this repo supports java8 lambdas? this is much cleaner with them

Reply:
Unfortunately, google-cloud-java is still at java 1.7 compatibility and we can't use lambdas.

Reply:
I wanted to use lambdas, but it's locked on java 7 here.